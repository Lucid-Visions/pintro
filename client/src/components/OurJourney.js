import React, { Component } from "react";
import { View, Text, StyleSheet } from "react-native";
import { LinearGradient } from "expo-linear-gradient";

class OurJourney extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  getDateFounded(dateFounded) {
    if (!dateFounded) {
      return null;
    }
    return (
      <View style={styles.experienceContainer}>
        <View style={styles.experienceCircle}></View>
        <Text style={{ fontFamily: "poppins-semi-bold" }}>
          Founded:{" "}
          <Text style={{ fontFamily: "poppins-medium", color: "grey" }}>
            {dateFounded}
          </Text>
        </Text>
      </View>
    );
  }

  getLocation(location) {
    if (!location) {
      return null;
    }
    return (
      <View style={styles.experienceContainer}>
        <View style={styles.experienceCircle}></View>
        <Text style={{ fontFamily: "poppins-semi-bold" }}>
          Location:{" "}
          <Text style={{ fontFamily: "poppins-medium", color: "grey" }}>
            {location}
          </Text>
        </Text>
      </View>
    );
  }

  getCompanySize(companySize) {
    if (!companySize) {
      return null;
    }
    return (
      <View style={styles.experienceContainer}>
        <View style={styles.experienceCircle}></View>
        <Text style={{ fontFamily: "poppins-semi-bold" }}>
          Company Size:{" "}
          <Text style={{ fontFamily: "poppins-medium", color: "grey" }}>
            {companySize+" Members"}
          </Text>
        </Text>
      </View>
    );
  }

  getFunding(funding) {
    if (!funding) {
      return null;
    }
    return (
      <View style={styles.experienceContainer}>
        <View style={styles.experienceCircle}></View>
        <Text style={{ fontFamily: "poppins-semi-bold" }}>
          Funding:{" "}
          <Text style={{ fontFamily: "poppins-medium", color: "grey" }}>
            {funding}
          </Text>
        </Text>
      </View>
    );
  }

  render() {
    const dateFounded = this.props.dateFounded;
    const location = this.props.location;
    const companySize = this.props.companySize;
    const funding = this.props.funding;

    return (
      <View>
        <Text
          style={{
            fontFamily: "poppins-semi-bold",
            marginTop: "4%",
            marginBottom: "2%"
          }}
        >
          Our Journey
        </Text>
        <View style={styles.experience}>
          <View style={styles.innerContainer}>
            <View style={styles.line}></View>
            {/* Date Founded */}
            {this.getDateFounded(dateFounded)}
            {/* Location */}
            {this.getLocation(location)}
            {/* Company Size */}
            {this.getCompanySize(companySize)}
            {/* Funding */}
            {this.getFunding(funding)}
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  experienceContainer: {
    flexDirection: "row",
    flexWrap: "wrap",
    alignItems: "center"
    // borderColor:"red",
    // borderWidth:1
  },
  subTextExperience: { 
    fontFamily: "poppins-medium", 
    color: "#757575" 
  },
  experienceCircle: {
    height: 7,
    width: 7,
    backgroundColor: "#F3A33F",
    borderRadius: 25,
    marginRight: 20,
    position: "absolute",
    left: -20,
    top: 7
  },
  indentOne: {
    marginLeft: 20
  },
  experience: {
    flexDirection: "row"
  },
  line: {
    backgroundColor: "#686868",
    width: 1,
    position: "absolute",
    left: -17,
    top: 10,
    bottom: 10
  },
  innerContainer: {
    marginLeft: 30
  }
});

export default OurJourney;
