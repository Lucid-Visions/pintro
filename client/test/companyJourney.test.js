import React from 'react'
import { create, act } from "react-test-renderer"
import OurJourney from '../src/components/OurJourney'

describe("<OurJourney />", () => {
    const props = {
        dateFounded: 15864314430000,
        location: "test center",
        companySize: 20,
        funding: "Looking for money"
    }
    test("snapshot of the component", () => {
        const component = create(<OurJourney {...props} />).toJSON()
        expect(component).toMatchSnapshot();
    })

    test("snapshot of the component with no info in props", () => {
        const component = create(<OurJourney />).toJSON()
        expect(component).toMatchSnapshot();
    })
})