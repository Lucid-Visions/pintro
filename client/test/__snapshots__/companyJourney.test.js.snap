// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OurJourney /> snapshot of the component 1`] = `
<View>
  <Text
    style={
      Object {
        "fontFamily": "poppins-semi-bold",
        "marginBottom": "2%",
        "marginTop": "4%",
      }
    }
  >
    Our Journey
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "marginLeft": 30,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#686868",
            "bottom": 10,
            "left": -17,
            "position": "absolute",
            "top": 10,
            "width": 1,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F3A33F",
              "borderRadius": 25,
              "height": 7,
              "left": -20,
              "marginRight": 20,
              "position": "absolute",
              "top": 7,
              "width": 7,
            }
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "poppins-semi-bold",
            }
          }
        >
          Founded:
           
          <Text
            style={
              Object {
                "color": "grey",
                "fontFamily": "poppins-medium",
              }
            }
          >
            15864314430000
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F3A33F",
              "borderRadius": 25,
              "height": 7,
              "left": -20,
              "marginRight": 20,
              "position": "absolute",
              "top": 7,
              "width": 7,
            }
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "poppins-semi-bold",
            }
          }
        >
          Location:
           
          <Text
            style={
              Object {
                "color": "grey",
                "fontFamily": "poppins-medium",
              }
            }
          >
            test center
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F3A33F",
              "borderRadius": 25,
              "height": 7,
              "left": -20,
              "marginRight": 20,
              "position": "absolute",
              "top": 7,
              "width": 7,
            }
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "poppins-semi-bold",
            }
          }
        >
          Company Size:
           
          <Text
            style={
              Object {
                "color": "grey",
                "fontFamily": "poppins-medium",
              }
            }
          >
            20 Members
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F3A33F",
              "borderRadius": 25,
              "height": 7,
              "left": -20,
              "marginRight": 20,
              "position": "absolute",
              "top": 7,
              "width": 7,
            }
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "poppins-semi-bold",
            }
          }
        >
          Funding:
           
          <Text
            style={
              Object {
                "color": "grey",
                "fontFamily": "poppins-medium",
              }
            }
          >
            Looking for money
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<OurJourney /> snapshot of the component with no info in props 1`] = `
<View>
  <Text
    style={
      Object {
        "fontFamily": "poppins-semi-bold",
        "marginBottom": "2%",
        "marginTop": "4%",
      }
    }
  >
    Our Journey
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "marginLeft": 30,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#686868",
            "bottom": 10,
            "left": -17,
            "position": "absolute",
            "top": 10,
            "width": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
